packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -1024,36 +1025,52 @@ export class Stack extends Construct implements ITaggable {
       return Fn.importValue(options.name);
     }
 
-    const resolvable = Tokenization.reverse(exportedValue);
-    if (!resolvable || !Reference.isReference(resolvable)) {
-      throw new Error('exportValue: either supply \'name\' or make sure to export a resource attribute (like \'bucket.bucketName\')');
-    }
-
-    // "teleport" the value here, in case it comes from a nested stack. This will also
-    // ensure the value is from our own scope.
-    const exportable = getExportable(this, resolvable);
+    const importValue = this.determineImportValue(exportedValue);
 
-    // Ensure a singleton "Exports" scoping Construct
-    // This mostly exists to trigger LogicalID munging, which would be
-    // disabled if we parented constructs directly under Stack.
-    // Also it nicely prevents likely construct name clashes
-    const exportsScope = getCreateExportsScope(this);
+    if (Array.isArray(importValue)) {
+      throw new Error('Attempted to export a list value from `exportValue()`: use `exportStringListValue()` instead');
+    }
```

Comment:
For some reason this makes me uneasy. Can you do inspections on the value's type and type hints, before we go to export it, instead?